storage/catalog.go

```diff
@@ -115,3 +116,18 @@ type StorageWithMetadata interface {
 type Metadata struct {
 	JobName string // name of the job using the storage
 }
+
+// ConnectAsDatabase returns a storage that is ready to use for reading and writing: `name` must corresponds to
+// a Database storage.
+func (c *Catalog) ConnectAsDatabase(ctx context.Context, name string, md Metadata) (*Database, error) {
+	strg, err := c.Connect(ctx, name, md)
+	if err != nil {
+		return nil, err
+	}
+
+	db, ok := strg.(*Database)
+	if !ok {
+		return nil, xerrors.Errorf("connecting to storage as database. unexpected type: %T", strg)
```

Comment:
Improve this error message since it's going to be shown to a user. Maybe `storage type is unsupported`. Log the type if you think it is useful.

Reply:
done.